app/scripts/metamask-controller.js

```diff
@@ -1291,6 +1291,21 @@ export default class MetamaskController extends EventEmitter {
       },
     );
 
+    this.controllerMessenger.subscribe(
+      `${this.snapController.name}:snapUpdated`,
+      (newSnap, oldVersion) => {
```

Comment:
[optional] it might be helpful to rename `newSnap` to `newTruncatedSnap` to preserve the type in the name